src/core/api/thread_ftd_api.cpp

```diff
@@ -273,22 +273,28 @@ otError otThreadGetChildInfoByIndex(otInstance *aInstance, uint16_t aChildIndex,
 
 otError otThreadGetChildNextIp6Address(otInstance *               aInstance,
                                        uint16_t                   aChildIndex,
-                                       otChildIp6AddressIterator *aIterator,
+                                       otChildIp6AddressIterator *aIterIndex,
```

Comment:
Should be still named `aIterator` since the type is still `otChildIp6AddressIterator`? 
API users don't need to know that the iterator is an index?

Reply:
The public definition in header file uses the `aIterator` (which is where API user would look for the documentation). 

This is the core internal implementation of this method. I renamed this to avoid confusion with `iter` variable and also make it easier and explicit (when reading the code) that it is being used as the index.

Reply:
Fine to use it for internal implementation.